lib/core/include/irods/connection_pool.hpp

```diff
@@ -15,6 +16,33 @@
 
 namespace irods
 {
+    /// Defines various options for changing the behavior of a connection pool.
+    ///
+    /// \since 4.3.1
+    struct connection_pool_options
+    {
+        /// Defines the number of API requests that can be sent via a connection before
+        /// refreshing it.
+        ///
+        /// Independent of any other option.
+        ///
+        /// \since 4.3.1
+        std::optional<std::int16_t> refresh_connections_after_n_retrievals;
+
+        /// Defines the number of seconds that must pass before refreshing a connection.
+        ///
+        /// Independent of any other option.
+        ///
+        /// \since 4.3.1
+        //std::optional<std::chrono::seconds> refresh_connections_after_n_seconds;
+
+        /// Instructs the connection pool to refresh a connection if changes to any resource
+        /// within the zone has been modified.
+        ///
+        /// \since 4.3.1
+        bool refresh_connections_after_resource_updates = false;
```

Comment:
I may change this name too.

Reply:
refresh_connections_when_resource_changes_detected

Reply:
Also ...
- `keep_resource_info_up_to_date`
- `track_resource_updates`
- `refresh_connections_if_resource_is_updated`
- `keep_resource_info_fresh`
- `watch_resource_modification_time`
- `track_resource_modification_time`
- `refresh_connections_following_resource_updates`

There's also the possibility of an enumeration.
```c++
enum class resource_info_sync_policy
{
    none,
    refresh_connections_always,

    // There's room for other policies to be supported.
};

struct connection_pool_options
{
    resource_info_sync_policy sync_policy = resource_info_sync_policy::none; // Default
};
```

Reply:
I like the enumeration idea. Would it be worth trying to communicate through the name that the connections are refreshed on retrieval (rather than, say, immediately)? Or perhaps that is all we can do and the user needs to understand that.

Reply:
Done